app/controllers/catalog_controller.rb

```diff
@@ -276,7 +276,15 @@ def index
       end
       (@response, @document_list) = get_search_results
 
-	  spelling_words = @response.spelling.words
+      spelling_words = []
+      @response.spelling.words.each do |word|
+        # handle Hash format for solr 5
```

Comment:
It'll work for both SOLR 4 and SOLR 5 as the spellcheck words will be converted to a string array for solr 4 and solr 5.